src/Data/Text/Utf8/AhoCorasick/Automaton.hs

```diff
@@ -239,53 +239,48 @@ type FallbackMap = IntMap State
 type ValuesMap v = IntMap [v]
 
 -- | Build the trie of the Aho-Corasick state machine for all input needles.
-buildTransitionMap :: forall v. [([CodeUnit], v)] -> (Int, TransitionMap, ValuesMap v)
-buildTransitionMap needles = buildTransitionMap' [(Utf8.decodeUtf8 cus, val) | (cus, val) <- needles]
-
-buildTransitionMap' :: forall v. [([CodePoint], v)] -> (Int, TransitionMap, ValuesMap v)
-buildTransitionMap' =
+buildTransitionMap :: forall v. [(Text, v)] -> (Int, TransitionMap, ValuesMap v)
```

Comment:
Did this have an effect on performance?

Reply:
It did consistently speed up the benchmarks, but only by a very small amount (in the order of magnitude of our measurement inaccuracies, i.e. some ten milliseconds).

Reply:
I still think it's nicer :D